lib/ash_admin/components/resource/form.ex

```diff
@@ -1043,9 +1043,9 @@ defmodule AshAdmin.Components.Resource.Form do
     end
   end
 
-  defp allow_nil_option(%{allow_nil?: true}), do: [{"", nil}]
-  defp allow_nil_option(%{default: nil}), do: [{"", nil}]
-  defp allow_nil_option(_), do: []
+  defp allow_nil_option(%{allow_nil?: true}), do: "Null"
+  defp allow_nil_option(%{default: default, allow_nil?: false}) when not is_nil(default), do: nil
+  defp allow_nil_option(_), do: "Select"
```

Comment:
So when an input is required, we show the option `Select` ? That won't be one of the options, right? So what does it look like in the UI? Could you add some screenshots of what the UX for the select will be with this change? Thanks for the fix!

Reply:
Oh, I just realized we removed the `allow_nil` option from the options list and we're just using it for the prompt. Doesn't that mean that you can't set the value to nil anymore? After changing it to a non-nil value I mean.

Reply:
> Oh, I just realized we removed the allow_nil option from the options list and we're just using it for the prompt. Doesn't that mean that you can't set the value to nil anymore? After changing it to a non-nil value I mean.
Exacly, currently it is impossible to set nil after you already have changed to non-nil.

> So when an input is required, we show the option Select ? That won't be one of the options, right? So what does it look like in the UI? Could you add some screenshots of what the UX for the select will be with this change?
`Select` will be still a option, but a invalid option.
I am adding a example with different scenarios